release/models/system/openconfig-system-grpc-tunnel.yang

```diff
@@ -0,0 +1,348 @@
+    tunnel (https://github.com/openconfig/grpctunnel/) to invoke one or
+    more dialout TCP over gRPC tunnels to bypass situations in which
+    direct dialin gRPC communications are not feasible.";
+
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision 2022-07-08 {
+    description
+      "Initial revision.";
+    reference "0.1.0";
+  }
+
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  typedef target-type {
+    type enumeration {
+      enum UNKNOWN {
+        value 0;
+        description
+          "An Unknown/Invalid target type";
+      }
+      enum SSH {
+        value 22;
+        description
+          "SSH protocol target type";
+      }
+      enum NETCONF_SSH {
+        value 830;
+        description
+          "NETCONF over SSH target type";
+      }
+      enum OPENFLOW {
+        value 6653;
+        description
+          "OpenFlow protocol target type";
+      }
+      enum GNMI_GNOI {
+        value 9339;
+        description
+          "gNMI/gNOI target type";
+      }
+      enum GRIBI {
+        value 9340;
+        description
+          "gRIBI target type";
+      }
+      enum P4RT {
+        value 9559;
+        description
+          "P4 Runtime target type";
+      }
+    }
+    description
+      "Sets the TCP based protocols to be tunneled over the gRPC tunnel";
+    reference
+      "OpenConfig gRPC Tunnel:
+      https://github.com/openconfig/grpctunnel/blob/master/proto/tunnel/tunnel.proto";
+  }
+
+  grouping tunnel-server-config {
+    description
+      "Configuration and state parameters corresponding to an individual
+      gRPC tunnel server";
+
+    leaf name {
+      type string;
+      description
+        "The name of the gRPC tunnel server instance";
+    }
+
+    leaf enable {
+      type boolean;
+      description
+        "When set to true, the tunnel server endpoint is enabled and an
+        outbound connection is initiated";
+    }
+
+    leaf host {
+      type oc-inet:host;
+      description
+        "The IPv4/IPv6 address or hostname of the tunnel server";
+    }
+
+    leaf port {
+      type oc-inet:port-number;
+      description
+        "The port number of the tunnel server";
+    }
+
+    leaf transport-security {
+      type boolean;
+      default true;
+      description
+        "Use gRPC transport security (TLS).  Enabled by default.  This
+        leaf allows transport security to be disabled for use cases that
+        are not supported, such as lab testing.";
+    }
+
+    leaf certificate-id {
+      type string;
+      description
+        "Name of the certificate that is associated with client (e.g.
+        mutual authentication).  The certificate ID is provisioned
+        through other interfaces to the device, such as the gNOI
+        certificate management service.  If this leaf is provisioned and
+        transport-security is enabled, mutual authentication is implied.
```

Comment:
Could you expand on the choice to implicitly turn on mutual authentication here? It seems like we might want to have this toggled individually -- i.e., I may use some other form of authentication for the session and not mutual cert auth whilst still wanting TLS.